.industrial_training_ci.bash

```diff
@@ -0,0 +1,272 @@
+  # actually run command
+  eval "${command}"
+  result=$?
+  ci_time_end
+  return $result
+}
+
+#######################################
+# Same as ci_run_impl() but silent command
+#
+# Arguments:
+#   command: action to run
+#######################################
+function ci_run_silent_impl() {
+  local command=$@
+
+  ci_time_start "${command} > /dev/null"
+  # actually run command
+  eval "${command} > /dev/null"
+  result=$?
+  ci_time_end
+  return $result
+}
+
+#######################################
+# Run a command and do timing for it
+#   Return the exit status of the command
+#######################################
+function ci_run() {
+  ci_run_impl $@ || exit $?
+}
+
+#######################################
+# Same as ci_run but return 0 exit status, thus ignoring any error
+#######################################
+function ci_run_true() {
+  ci_run_impl $@ || return 0
+}
+
+#######################################
+# Same as ci_run but silent output
+#######################################
+function ci_run_silent() {
+  ci_run_silent_impl $@ || exit $?
+}
+
+function setup()
+{
+  ci_run_silent apt update -y
+  ci_run_silent apt install python-catkin-tools -y
+  ci_run_silent apt install ros-$ROS_DISTRO-moveit -y
+  ci_run_silent apt install ros-$ROS_DISTRO-pcl-ros -y
+  ci_run_silent apt upgrade -y
+}
+
+######################################################
+## Build a catkin workspace with rosdep and wstool ##
+######################################################
+function build_ws()
+{
+  ci_run source /opt/ros/$ROS_DISTRO/setup.bash
+  ci_run cd $HOME
+
+  # Make a new catkin ws
+  ci_run mkdir catkin_ws
+  ci_run cd catkin_ws
+
+  # Copy the exercise workspace into the home directory
+  ci_run cp -r $TRAVIS_BUILD_DIR/$1 .
+
+  # Initialize the workspace
+  ci_run catkin init
+
+  # Perform rosinstall
+  ci_run wstool init
+  if [ -f src/.rosinstall ]; then
+    ci_run wstool merge -t src/ src/.rosinstall
+    ci_run wstool update -t src
+  fi
+
+  # Install dependencies with rosdep
+  ci_run_silent rosdep -q install --from-paths src --ignore-src -r -y
+
+  # Build the workspace
+  ci_run catkin build --no-status --summarize
+}
+
+################################################################################
+## Update the container image to save time later when installing dependencies ##
+################################################################################
+function update_image()
+{
+  # Run the docker container
+  ci_run docker run --name $CONTAINER_NAME --network=host -d -i \
+    -v $TRAVIS_BUILD_DIR:$TRAVIS_BUILD_DIR \
+    -e TRAVIS_BUILD_DIR=$TRAVIS_BUILD_DIR \
+    $CONTAINER_IMAGE
+
+  # Source this file in the docker container and run the setup routine
+  ci_run docker exec -t $CONTAINER_NAME \
+    bash -c \"source $TRAVIS_BUILD_DIR/.industrial_training_ci.bash\; ci_run setup\"
+
+  # Create a new name for the updated image
+  ci_run export NEW_CONTAINER_IMAGE=$CONTAINER_IMAGE\_update
```

Comment:
This doesn't feel right since it's writing what's essentially a global variable in one function that another function makes use of. Could arguments to the bash functions be used to avoid this? Or is it even necessary to generate a new container name and can the same one be used after `docker commit`?